docs/docs/spec.md

```diff
@@ -379,20 +384,9 @@ condition is met, skip the remaining conditions.
 > `@Nullable`, then the override's return type will *not* have nullness operator
 > `UNION_NULL`.
 
-> The rules here never produce the fourth nullness operator, `MINUS_NULL`.
-> However, if tool authors prefer, they can safely produce `MINUS_NULL` in any
-> case in which it is equivalent to `NO_CHANGE`. For example, there is no
-> difference between `String NO_CHANGE` and `String MINUS_NULL`.
-
-> So why does `MINUS_NULL` exist at all? It does appear later in this spec in
-> the section on [substitution]. However, its main purpose is to provide tools
-> with a way to represent the nullness of certain expressions in implementation
-> code: Consider `ArrayList<E>`. `ArrayList` supports null elements, so the
-> class has to handle the possibility that any expression of type `E` may be
-> null. However, if implementation code contains the statement `if (e != null) {
-> ... }`, then tools can assume that `e` is non-null inside. The purpose of
-> `MINUS_NULL` is to represent that such an expression is known not to be null,
-> even though its base type `E` suggests otherwise.
+> If tool authors prefer, they can safely produce `MINUS_NULL` in any case in
+> which it is equivalent to `NO_CHANGE`. For example, there is no difference
```

Comment:
Is that just non-type-variable usages?

Reply:
Doing so for intersection types would also be a problem (and I guess maybe union types if we got into those, which maybe we should now that we're getting into implementation code more: #621).

It's probably just those two (or three). Worth mentioning?

Reply:
If you have a specific idea of which cases you mean, please list them. But list the smaller set. Is it shorter to list type variables and intersection/union types?

Reply:
They might just be the same length :) Have a look at the two lists I've included, and see what you think about which one(s) to keep.

(See https://github.com/jspecify/jspecify/issues/621 for a relevant note about union types.)